mllib/src/main/scala/org/apache/spark/ml/feature/ChiSqSelector.scala

```diff
@@ -291,9 +291,13 @@ final class ChiSqSelectorModel private[ml] (
     val featureAttributes: Array[Attribute] = if (origAttrGroup.attributes.nonEmpty) {
       origAttrGroup.attributes.get.zipWithIndex.filter(x => selector.contains(x._2)).map(_._1)
     } else {
-      Array.fill[Attribute](selector.size)(NominalAttribute.defaultAttr)
+      null
```

Comment:
Instead of removing this, I think the better way is fill the `values` or `numValues` member here.

Reply:
This function is in ChiSqSelectorModel, in the model you only have "selectedFeatures: Array[Int]", seems you cannot fill values and or numValues only with the model information.

Reply:
For this problem, either here or the following code should be changed, right? 
 " case nomAttr: NominalAttribute =>
              nomAttr.getNumValues match {
                case Some(numValues: Int) => Iterator(idx -> numValues)
                case None => throw new IllegalArgumentException(s"Feature $idx is marked as" +
                  " Nominal (categorical), but it does not have the number of values specified.")
              }"

Reply:
Yes I admit it is hard to get the `values` and/or `numValues` here.
Current spark code will throw exception when we pipeline ChiSqSelector + DecisionTreeClassifier (on features without attributes),
But if we remove the code adding `Nominal` here (in ChiSqSelector), although pipeline ChiSqSelector + DecisionTreeClassifier(on features without attributes) will run successfully, it will get wrong result, because DecisionTreeClassifier treat them as continuous features.
Comparing explicitly throwing exception and running successfully on the surface(but internal is wrong), I tend to keep current choice.

Reply:
The problem is should we allow add Nominal and not set value or numvalues. If we allow this,  should we change the code of getCategoricalFeatures?

Reply:
So I suggest keep current code, but you can improve the error description here https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/util/MetadataUtils.scala#L66:
```
 throw new IllegalArgumentException(s"Feature $idx is marked as" +
" Nominal (categorical), but it does not have the number of values specified.")
```
Add tips here, tell user it can use `VectorIndexer` to process categorical features first, to solve this issue.

Reply:
Change the error description is ok, but ChiSqSelector maybe not the only case causes this error. StringIndexer, QuantileDiscretizer may also causes this error. 
Maybe we need a solution for this problem?